### PR TITLE
[border agent] [WIP] publish meshcop service when thread is disabled (part 1/3): Add otBorderAgentStart() API

### PR DESCRIPTION
**Pull request by superwhd**
This PR is part 1 of the fix of https://github.com/openthread/ot-br-posix/issues/736.

Part 1: https://github.com/openthread/openthread/pull/6847
Part 2: https://github.com/openthread/ot-br-posix/pull/942
Part 3: https://github.com/openthread/openthread/pull/6850 (locally tested, to be updated after https://github.com/openthread/openthread/pull/6865 is merged)

Depends-On: openthread/ot-br-posix#942

### CODE REVIEW

File: src/posix/platform/system.cpp [hunk 1]
```diff
@@ -152,6 +153,10 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
 #if OPENTHREAD_POSIX_CONFIG_DAEMON_ENABLE
     ot::Posix::Daemon::Get().Enable(instance);
 #endif
+
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+    otBorderAgentStart(instance);
```
Comment (wgtdkp) [inline src/posix/platform/system.cpp:158]: Stop the Border Agent in `otSysDeinit`?

File: src/posix/platform/system.cpp [hunk 2]
```diff
@@ -152,11 +155,18 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
...
+#endif
+    return sInstance = instance;
 }
 
 void otSysDeinit(void)
 {
+#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
+    otBorderAgentStop(sInstance);
+#endif
```
Comment (wgtdkp) [inline src/posix/platform/system.cpp:169]:
```suggestion
void otSysDeinit(otInstance *aInstance)
{
#if OPENTHREAD_CONFIG_BORDER_ROUTER_ENABLE && OPENTHREAD_CONFIG_PLATFORM_NETIF_ENABLE
    otBorderAgentStop(aInstance);
#endif
```

**Review by wgtdkp — APPROVED** [review]
LGTM 👍

File: src/posix/platform/system.cpp (hunk 2 above)
Comment (superwhd) [reply src/posix/platform/system.cpp:169]: Doing so will break the build of ot-br-posix. Any shall we introduce a new API like `otSysDeinitForInstance(otInstance *)`?

**Comment by wgtdkp** [comment]
How about starting the Border Agent in `Instance::AfterInit()`?

File: src/posix/platform/system.cpp (hunk 2 above)
Comment (superwhd) [reply src/posix/platform/system.cpp]: Resolved via offline discussion.

**Pull request closed without merging**